velox/functions/lib/aggregates/SetBaseAggregate.h

```diff
@@ -50,7 +53,15 @@ class SetBaseAggregate : public exec::Aggregate {
     for (auto i = 0; i < numGroups; ++i) {
       auto* group = groups[i];
       if (isNull(group)) {
-        arrayVector->setNull(i, true);
+        if constexpr (nullForEmpty) {
+          arrayVector->setNull(i, true);
+        } else {
+          // If the group's accumulator is null, the corresponding result is an
+          // empty array.
+          clearNull(rawNulls, i);
+          rawOffsets[i] = numValues;
```

Comment:
nit: It might be clearer to set this to 0. This is how we do that elsewhere in the code: empty and NULL entries have offset == size == 0.

Reply:
Updated. Thanks for the review.